java-frontend/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -447,4 +490,41 @@ public String toString() {
     }
   };
 
+  private static class Equals implements BiConsumer {
+    private final AVLTree tree;
+    private int sizeDifference;
+
+    public static boolean compute(AVLTree first, AVLTree second) {
+      Equals state = new Equals(first);
+      if (!state.supersetOf(second)) {
+        return false;
+      }
+      first.forEach(state);
+      return state.sizeDifference == 0;
+    }
+
+    private Equals(AVLTree tree) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1213)